### PR TITLE
Added a note about scientific notation strings

### PR DESCRIPTION
**Pull request by machavity**
There's nothing explicit in the manual about scientific notation strings being interpreted as number. A section that people can be pointed to might be helpful

### CODE REVIEW

File: language/types/numeric-strings.xml [hunk 1]
```diff
@@ -30,6 +30,22 @@ NUM_STRING       ({INT_NUM_STRING} | {FLOAT_NUM_STRING})
   This is simply a string which starts like a numeric string followed by
   any characters.
  </para>
+ 
+   <note>
+    <para>Any string that contains the letter <literal>E</literal> (case insensitive)
```
Comment (cmb69) [inline language/types/numeric-strings.xml:35]: I suggest to break the line after the `<para>` for better readability:
```suggestion
    <para>
     Any string that contains the letter <literal>E</literal> (case insensitive) 
```

**Review by cmb69 — COMMENTED** [review]
Thank you for the PR! I'm not sure, though, whether that needs to be explicitly documented; there is already the grammar above which specifies that "e/E" can be part of a numeric string, and there is an example below. I'd like to hear what others think about this.

**Comment by Girgias** [comment]
I think it would be better to have an example section, where one of them is in exponential notation, as the ``e`` specifier is already present just not very visible.

**Comment by machavity** [comment]
I'm fine with either. It's just one of those arcane gotchas of PHP and the manual gives examples (there's some given without explanation at https://www.php.net/manual/en/language.operators.comparison.php but it didn't seem to fit to add any notes there). Was trying to address this question and was surprised the manual has nothing direct https://stackoverflow.com/questions/72126877/illogical-string-comparison-in-php

**Comment by Crell** [comment]
I think I'd favor an example, but either works for me.  @machavity are you going to continue this?

**Comment by machavity** [comment]
What would I need to do to continue? I can make the readability edit cmb mentions if that's all.

**Comment by cmb69** [comment]
> I can make the readability edit cmb mentions if that's all.

Yes, please do that. And also have a look at the indentation; apparently the new lines have 2 spaces too much.

File: language/types/numeric-strings.xml [hunk 2]
```diff
@@ -30,6 +30,23 @@ NUM_STRING       ({INT_NUM_STRING} | {FLOAT_NUM_STRING})
   This is simply a string which starts like a numeric string followed by
   any characters.
  </para>
+ 
+ <note>
+  <para>
+   Any string that contains the letter <literal>E</literal> (case insensitive) 
+   bounded by numbers will be seen as a number expressed in scientific notation.
```
Comment (cmb69) [inline language/types/numeric-strings.xml:37]: These lines have trailing whitespace:
```suggestion
   Any string that contains the letter <literal>E</literal> (case insensitive)
   bounded by numbers will be seen as a number expressed in scientific notation.
```

File: language/types/numeric-strings.xml [hunk 3]
```diff
@@ -30,6 +30,23 @@ NUM_STRING       ({INT_NUM_STRING} | {FLOAT_NUM_STRING})
   This is simply a string which starts like a numeric string followed by
   any characters.
  </para>
+
```
Comment (cmb69) [inline language/types/numeric-strings.xml:33]:
```suggestion

```

**Comment by cmb69** [comment]
Oh, there actually was a line with a single space; I've fixed that and committed.

Thanks for the PR again!

**Pull request closed without merging**